main/forum/forumfunction.inc.php

```diff
@@ -4160,8 +4160,8 @@ function store_edit_post($forumInfo, $values)
 
     // Update the post_title and the post_text.
     $params = [
-        'post_title' => $values['post_title'],
-        'post_text' => $values['post_text'],
+        'post_title' => Security::remove_XSS($values['post_title']),
```

Comment:
The same here, Security::remove_XSS should be added before displaying the content.